### PR TITLE
feat: Add nullable vector support in import utility layer

### PR DESCRIPTION
**Pull request by marcelo-cjl**
related: #45993 

Add nullable vector support in import utility layer
    
Key changes:

ImportV2 util:
- Add nullable vector types (FloatVector, Float16Vector, BFloat16Vector,
  BinaryVector, SparseFloatVector, Int8Vector) to AppendNullableDefaultFieldsData()
- Add tests for nullable vector field data appending

CSV/JSON/Numpy readers:
- Add nullPercent parameter to test data generation for better null coverage
- Mark vector fields as nullable in test schemas
- Add test cases for nullable vector field parsing
- Refactor tests to use loop-based approach with 0%, 50%, 100% null percentages

Parquet field reader:
- Add ReadNullableBinaryData() for nullable BinaryVector/Float16Vector/BFloat16Vector
- Add ReadNullableFloatVectorData() for nullable FloatVector
- Add ReadNullableSparseFloatVectorData() for nullable SparseFloatVector
- Add ReadNullableInt8VectorData() for nullable Int8Vector
- Add ReadNullableStructData() for generic nullable struct data
- Update Next() to use nullable read methods when field is nullable
- Add null data validation for non-nullable fields

- Core invariant: import must preserve per-row alignment and validity for every field — nullable vector fields are expected to be encoded with per-row validity masks and all readers/writers must emit arrays aligned to original input rows (null entries represented explicitly).
- New feature & scope: adds end-to-end nullable-vector support in the import utility layer — AppendNullableDefaultFieldsData in internal/datanode/importv2/util.go now appends nil placeholders for nullable vectors (FloatVector, Float16Vector, BFloat16Vector, BinaryVector, SparseFloatVector, Int8Vector); parquet reader (internal/util/importutilv2/parquet/field_reader.go) adds ReadNullableBinaryData, ReadNullableFloatVectorData, ReadNullableSparseFloatVectorData, ReadNullableInt8VectorData, ReadNullableStructData and routes nullable branches to these helpers; CSV/JSON/Numpy readers and test utilities updated to generate and validate 0/50/100% null scenarios and mark vector fields as nullable in test schemas.
- Logic removed / simplified: eliminates ad-hoc "parameter-invalid" rejections for nullable vectors inside FieldReader.Next by centralizing nullable handling into ReadNullable* helpers and shared validators (getArrayDataNullable, checkNullableVectorAlignWithDim/checkNullableVectorAligned), simplifying control flow and removing scattered special-case checks.
- No data loss / no regression (concrete code paths): nulls are preserved end-to-end — AppendNullableDefaultFieldsData explicitly inserts nil entries per null row (datanode import append path); ReadNullable*Data helpers return both data and []bool validity masks so callers in field_reader.go and downstream readers receive exact per-row validity; testutil.BuildSparseVectorData was extended to accept validData so sparse vectors are materialized only for valid rows while null rows are represented as missing. These concrete paths ensure null rows are represented rather than dropped, preventing data loss or behavioral regression.

### CODE REVIEW

File: internal/datanode/importv2/util_test.go [hunk 1]
```diff
@@ -599,8 +641,21 @@ func Test_AppendNullableDefaultFieldsData(t *testing.T) {
...
+					assert.Equal(t, count, len(validData), "ValidData length should equal row count")
+					for _, v := range validData {
+						assert.False(t, v, "all values should be null")
+					}
+					isVectorType := tt.dataType == schemapb.DataType_FloatVector ||
+						tt.dataType == schemapb.DataType_Float16Vector ||
+						tt.dataType == schemapb.DataType_BFloat16Vector ||
+						tt.dataType == schemapb.DataType_BinaryVector ||
+						tt.dataType == schemapb.DataType_SparseFloatVector ||
+						tt.dataType == schemapb.DataType_Int8Vector
```
Comment (foxspy) [inline internal/datanode/importv2/util_test.go]: The GetRow test shouldn’t be skipped.

File: internal/util/importutilv2/parquet/field_reader.go [hunk 2]
```diff
@@ -1430,6 +1741,72 @@ func ReadNullableIntegerOrFloatArrayData[T constraints.Integer | constraints.Flo
 	return data, validData, nil
 }
 
+func ReadNullableFloatVectorData(pcr *FieldReader, count int64) (any, []bool, error) {
```
Comment (foxspy) [inline internal/util/importutilv2/parquet/field_reader.go:1641]: How are null vectors represented inside a Parquet file? Please add some comments here to explain this.

File: internal/util/importutilv2/parquet/field_reader.go [hunk 3]
```diff
@@ -1045,6 +1193,122 @@ func ReadSparseFloatVectorData(pcr *FieldReader, count int64) (any, error) {
...
+		return nil, nil, err
+	}
+	validCount := 0
+	for _, v := range validData {
+		if v {
+			validCount++
+		}
+	}
+	return &storage.SparseFloatVectorFieldData{
+		SparseFloatArray: schemapb.SparseFloatArray{
```
Comment (foxspy) [inline internal/util/importutilv2/parquet/field_reader.go:1199]: validData is being returned twice here

File: internal/util/importutilv2/parquet/field_reader.go [hunk 4]
```diff
@@ -329,6 +329,18 @@ func ReadNullableBoolData(pcr *FieldReader, count int64) (any, []bool, error) {
 		defaultValue := pcr.field.GetDefaultValue().GetBoolData()
 		return fillWithDefaultValueImpl(data, defaultValue, validData, pcr.field)
 	}
+	if !pcr.field.GetNullable() {
+		for _, valid := range validData {
```
Comment (foxspy) [inline internal/util/importutilv2/parquet/field_reader.go]: duplicate code

File: internal/datanode/importv2/util_test.go (hunk 1 above)
Comment (marcelo-cjl) [reply internal/datanode/importv2/util_test.go]: done

File: internal/util/importutilv2/parquet/field_reader.go (hunk 4 above)
Comment (marcelo-cjl) [reply internal/util/importutilv2/parquet/field_reader.go]: have been deleted

File: internal/util/importutilv2/parquet/field_reader.go (hunk 3 above)
Comment (marcelo-cjl) [reply internal/util/importutilv2/parquet/field_reader.go:1199]: this is mainly to maintain the consistency of function interface

File: internal/util/importutilv2/parquet/field_reader.go (hunk 2 above)
Comment (marcelo-cjl) [reply internal/util/importutilv2/parquet/field_reader.go:1641]: In Parquet, nullable vectors are stored as LIST type (e.g., list<float32>). Null values are tracked using definition levels - a value of 0 means null, higher values mean valid data. When read via Apache Arrow, check listReader.IsNull(i) per row.

**Comment by bigsheeper** [comment]
/lgtm
lgtm for the import part

**Comment by marcelo-cjl** [comment]
feature: #45993

**Review by czs007 — APPROVED** [review]
/lgtm
